app/controllers/amr_heatmap_controller.rb

```diff
@@ -0,0 +1,39 @@
+class AmrHeatmapController < ApplicationController
+  include ReportHelper
+
+  before_action :admin_required
+
+  power :viewable_samples
+
+  # GET /amr_heatmap.json
+  # Return JSON information required to create a visualization,
+  # from submitted parameters
+  def amr_counts
+    samples = []
+    amr_data = []
+
+    params[:id].each do |param_id|
+      sample = current_power.viewable_samples.find(param_id)
+      assert_access
```

Comment:
I'm honestly not 100% sure what the point of `assert_access` is (it's in `application_controller`), but you shouldn't need to do it here.

Reply:
Removed!